doc/contribute/coding_guidelines/index.rst

```diff
@@ -1297,6 +1297,191 @@ macros. Given that Zephyr uses a fork of the corresponding upstream for each
 module, it is always possible to patch the macro implementation in each module
 to avoid collisions.
 
+.. _coding_guideline_libc_usage_restrictions_in_zephyr_kernel:
+
+Rule A.4: C Standard Library Usage Restrictions in Zephyr Kernel
+================================================================
+
+Severity
+--------
+
+Required
+
+Description
+-----------
+
+The use of the C standard library functions and macros in the Zephyr kernel
+shall be limited to the following functions and macros from the ISO/IEC
+9899:2011 standard, also known as C11, and their extensions:
+
+.. csv-table:: List of allowed libc functions and macros in the Zephyr kernel
+   :header: Function,Source
+   :widths: auto
+
+   abort(),ISO/IEC 9899:2011
```

Comment:
You've defined C11 as an alias for ISO/IEC 9899:2011, might be nice to use the shorter name in this table?

Reply:
Given that there is plenty of horizontal space left when this table gets rendered, I would prefer to keep the full name.

It would also be nicer to use the full name here from the consistency perspective in case we ever need to reference other standards in the future.